### PR TITLE
Fix: Provide Linux amd64 tar.gz distribution #179

### PR DESCRIPTION
**Pull request by varshashesh**
# Description of the issue
This is a fix for the issue #179 

# Description of changes
Saving tar file before creating rpm in Tools/src/create_rpm.sh

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
make
make release 

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make linter`

### CODE REVIEW

File: Tools/src/create_rpm.sh
Comment (SaxyPandaBear) [inline Tools/src/create_rpm.sh:52]: I think for consistency, we should remove the extra echos here. The 
```bash
echo "*************************************************"
```
usage is only at the beginning of the script.

File: Tools/src/create_rpm.sh
Comment (SaxyPandaBear) [inline Tools/src/create_rpm.sh:54]: I kind of want to move this `ls` call to the bottom after the rpm copy, but I'm not sure right now. I'll think about it more

**Review by SaxyPandaBear — COMMENTED** [review]
Thanks for the contribution! Overall looks pretty good to me. Just some minor comments. 

Separately, it would be great if you could show output of `make package-rpm` that shows the new `ls` output, and if you could unzip the persisted artifact to verify its contents. 

We'd still have to do work on our end in order to start publishing it to S3 but this is a great first step

File: Tools/src/create_rpm.sh
Comment (varshashesh) [reply Tools/src/create_rpm.sh:52]: Files in the extracted tar file: [ec2-user@ip-172-31-1-251 del]$ tar -xvf ../amazon-cloudwatch-agent.tar.gz 
opt/
opt/aws/
opt/aws/amazon-cloudwatch-agent/
opt/aws/amazon-cloudwatch-agent/logs/
opt/aws/amazon-cloudwatch-agent/var/
opt/aws/amazon-cloudwatch-agent/etc/
opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.d/
opt/aws/amazon-cloudwatch-agent/etc/common-config.toml
opt/aws/amazon-cloudwatch-agent/bin/
opt/aws/amazon-cloudwatch-agent/bin/CWAGENT_VERSION
opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent
opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl
opt/aws/amazon-cloudwatch-agent/bin/config-translator
opt/aws/amazon-cloudwatch-agent/bin/config-downloader
opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-config-wizard
opt/aws/amazon-cloudwatch-agent/bin/start-amazon-cloudwatch-agent
opt/aws/amazon-cloudwatch-agent/doc/
opt/aws/amazon-cloudwatch-agent/doc/amazon-cloudwatch-agent-schema.json
opt/aws/amazon-cloudwatch-agent/LICENSE
opt/aws/amazon-cloudwatch-agent/NOTICE
opt/aws/amazon-cloudwatch-agent/THIRD-PARTY-LICENSES
opt/aws/amazon-cloudwatch-agent/RELEASE_NOTES
etc/
etc/init/
etc/init/amazon-cloudwatch-agent.conf
etc/systemd/
etc/systemd/system/
etc/systemd/system/amazon-cloudwatch-agent.service

File: Tools/src/create_rpm.sh
Comment (khanhntd) [inline Tools/src/create_rpm.sh:53]: It would be better to provide a constant since we use in multiple different places. Therefore, avoid human error and easier in reading.

**Review by khanhntd — COMMENTED** [review]
As Andrew said, thanks for contribution we appreciated it !!!!. Overall for the code, LGTM. There is only a minor comment from me. However, would be good to show the output and put it in the PR (e.g `make release` and inside the zip artifacts with the folder its belong to) for future reference without looking at the comments. Again its a first step so we appreciate your contribution !!!!

**Review by khanhntd — APPROVED** [review]

**Comment by varshashesh** [comment]
Thanks for the approval @khanhntd :)

**Comment by varshashesh** [comment]
I need another review to merge  this. 
I also see merging is blocked because of build-and-test-macos / macos-unittest (pull_request)
My changes are only on linux which passes the build and test

Please comment.

**Comment by varshashesh** [comment]
> This PR was marked stale due to lack of activity.

I need another review - At least 2 approving reviews are required by reviewers with write access
Can you help me with this  build-and-test-macos / macos-unittest Failling

**Comment by varshashesh** [comment]
Is the CI failure similar to this bug reported: https://github.com/aws/amazon-cloudwatch-agent/issues/553

**Comment by khanhntd** [comment]
Hi @varshashesh,
For the build-and-test-macos, yes the attached issues are the right one. However, it is not a block for the PR to get merged. For awareness, there are some internal discussions regarding support TAR for Linux (aka this PR)  so we will merge the PR after final decision has been made and thanks for the wait.

**Comment by chadpatel** [comment]
No plans to release tar.gz release at this time

**Pull request closed without merging**
